front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -76,8 +88,55 @@ class IntentMenu extends Component {
         );
     }
 
+    starAndFlagFilter = (intents) => {
+        const filteredIntents = [];
+
+        for (let intent of intents) {
+            if (this.state.starFilter) {
+                if (intent.star) {
+                    filteredIntents.push(intent);
+                    continue;
+                }
+            }
+            if (this.state.flagFilter) {
+                if (intent.flag) {
+                    filteredIntents.push(intent);
+                }
+            }
+        }
+        return filteredIntents;
+    }
+
+    addStar = (intent) => {
```

Comment:
It might be better to name this a little more descriptively. Maybe something like addStarIndicator